### PR TITLE
add --latest mode for replicate-logs

### PR DESCRIPTION
**Pull request by robinbryce**
this will automaticlaly run the watch to get all relevant changes.

This was requested by Jim Wilburly based on customer and partner feedback

### CODE REVIEW

**Comment by robinbryce** [comment]
test evidence for systemtests on macos

banks-7:veracity robin$ ta build:fast
banks-7:veracity robin$ ls -la ./veracity
-rwxr-xr-x@ 1 robin  staff  15798594 Nov  5 09:20 ./veracity
banks-7:veracity robin$ date
Tue Nov  5 09:20:37 GMT 2024
banks-7:veracity robin$ ta test:system
/Users/robin/wb/rkvst/veracity/tests/systemtest /Users/robin/wb/rkvst/veracity
testVeracityVersion
testVeracityWatchPublicFindsActivity
testVeracityReplicateLogsPublicTenantWatchPipe
testVeracityReplicateLogsPublicTenantWatchLatestFlag
testVerifySingleEvent
testVerifyListEvents
testVerifySingleEventWithLocalMassifCopy
testVerifyIncludeLocalErrorForDuplicateMassifs
testVerboseOutput
testHelpOutputNoArgs
testValidEventNotinMassif
Expected (hex):
00000000  65 72 72 6f 72 3a 20 74  68 65 20 65 6e 74 72 79  |error: the entry|
00000010  20 69 73 20 6e 6f 74 20  69 6e 20 74 68 65 20 6c  | is not in the l|
00000020  6f 67 2e 20 66 6f 72 20  74 65 6e 61 6e 74 20 74  |og. for tenant t|
00000030  65 6e 61 6e 74 2f 36 65  61 35 63 64 30 30 2d 63  |enant/6ea5cd00-c|
00000040  37 31 31 2d 33 36 34 39  2d 36 39 31 34 2d 37 62  |711-3649-6914-7b|
00000050  31 32 35 39 32 38 62 62  62 34 0a                 |125928bbb4.|
0000005b
Actual (hex):
00000000  65 72 72 6f 72 3a 20 74  68 65 20 65 6e 74 72 79  |error: the entry|
00000010  20 69 73 20 6e 6f 74 20  69 6e 20 74 68 65 20 6c  | is not in the l|
00000020  6f 67 2e 20 66 6f 72 20  74 65 6e 61 6e 74 20 74  |og. for tenant t|
00000030  65 6e 61 6e 74 2f 36 65  61 35 63 64 30 30 2d 63  |enant/6ea5cd00-c|
00000040  37 31 31 2d 33 36 34 39  2d 36 39 31 34 2d 37 62  |711-3649-6914-7b|
00000050  31 32 35 39 32 38 62 62  62 34 0a                 |125928bbb4.|
0000005b
testNon200Response
testMissingMassifFile
testNotBlobFile
testInvalidBlobUrl
Expected (hex):
00000000  65 72 72 6f 72 3a 20 75  6e 65 78 70 65 63 74 65  |error: unexpecte|
00000010  64 20 65 6e 64 20 6f 66  20 4a 53 4f 4e 20 69 6e  |d end of JSON in|
00000020  70 75 74 0a                                       |put.|
00000024
Actual (hex):
00000000  65 72 72 6f 72 3a 20 75  6e 65 78 70 65 63 74 65  |error: unexpecte|
00000010  64 20 65 6e 64 20 6f 66  20 4a 53 4f 4e 20 69 6e  |d end of JSON in|
00000020  70 75 74 0a                                       |put.|
00000024

Ran 15 tests.

JUnit XML file res.xml was saved.

OK

**Review by honourfish — APPROVED** [review]

File: replicatelogs.go [hunk 1]
```diff
@@ -119,7 +130,7 @@ By default transient errors are re-tried without limit, and if the error is 429,
 				return err
 			}
 
-			changes, err := readTenantMassifChanges(cCtx)
+			changes, err := readTenantMassifChanges(context.TODO(), cCtx, cmd)
```
Comment (wp0pw) [inline replicatelogs.go:133]: guessing we don't really have a better context here? maybe short comment to explain what the plan for this one is?

File: replicatelogs.go [hunk 2]
```diff
@@ -624,3 +604,86 @@ func peakBaggedRoot(state massifs.MMRState) []byte {
 	}
 	return mmr.HashPeaksRHS(sha256.New(), state.Peaks)
 }
+
+type changeCollector struct {
+	log         logger.Logger
+	watchOutput string
+}
+
+func (c *changeCollector) Logf(msg string, args ...any) {
```
Comment (wp0pw) [inline replicatelogs.go:613]: are we actually log through collector?

Comment (robinbryce) [reply replicatelogs.go:613]: > are we actually log through collector?

Yeah, that is what the watch command does. It reports as it goes. I could do a separate change to revisit that. Or if you're his is controversial I can kick this or from the release. Jim has been waiting a few weeks now tho

Comment (robinbryce) [reply replicatelogs.go:613]: Please reopen if you want me to change something

**Review by wp0pw — APPROVED** [review]

**Pull request merged**